pkg/daemon/daemon.go

```diff
@@ -303,86 +308,108 @@ func ptp4lCreateCmd(nodeProfile *ptpv1.PtpProfile, confFilePath string) *exec.Cm
-	}()
-
-	//
-	// don't discard process stderr output
-	//
-	p.cmd.Stderr = os.Stderr
-	cmdReader, err := p.cmd.StdoutPipe()
-	if err != nil {
-		glog.Errorf("cmdRun() error creating StdoutPipe for %s: %v", p.name, err)
-		return
-	}
-
-	done := make(chan struct{})
-
-	if !stdoutToSocket {
-		scanner := bufio.NewScanner(cmdReader)
-		go func() {
-			for scanner.Scan() {
-				output := scanner.Text()
-				fmt.Printf("%s\n", output)
-				extractMetrics(p.configName, p.name, p.ifaces, output)
-			}
-			done <- struct{}{}
-		}()
-	} else {
-		go func() {
-		connect:
-			select {
-			case <-p.exitCh:
-				done <- struct{}{}
-			default:
-				c, err = net.Dial("unix", eventSocket)
-				if err != nil {
-					glog.Errorf("error trying to connect to event socket")
-					time.Sleep(connectionRetryInterval)
-					goto connect
-				}
-			}
+		if !stdoutToSocket {
 			scanner := bufio.NewScanner(cmdReader)
-			for scanner.Scan() {
-				output := scanner.Text()
-				out := fmt.Sprintf("%s\n", output)
-				fmt.Printf("%s", out)
-				_, err := c.Write([]byte(out))
-				if err != nil {
-					glog.Errorf("Write error:", err)
-					goto connect
+			go func() {
+				for scanner.Scan() {
+					output := scanner.Text()
+					fmt.Printf("%s\n", output)
+					extractMetrics(p.configName, p.name, p.ifaces, output)
+				}
+				done <- struct{}{}
+			}()
+		} else {
+			go func() {
+			reconnect:
+				for {
+					p.restartMutex.Lock() // Don't restart after termination
+					restart := p.restart
+					p.restartMutex.Unlock()
+					if restart {
+						c, err = net.Dial("unix", eventSocket)
+						if err != nil {
+							glog.Errorf("error trying to connect to event socket")
+							time.Sleep(connectionRetryInterval)
+							continue
+						}
+					} else {
+						break
+					}
+
+					scanner := bufio.NewScanner(cmdReader)
+					for scanner.Scan() {
+						output := scanner.Text()
+						out := fmt.Sprintf("%s\n", output)
+						fmt.Printf("%s", out)
+						_, err := c.Write([]byte(out))
+						if err != nil {
+							glog.Errorf("Write error:", err)
+							continue reconnect
+						}
+					}
+					break
 				}
+				done <- struct{}{}
+			}()
+		}
+		p.restartMutex.Lock() // Don't restart after termination
+		restart := p.restart
+		skipWait := false
+		if restart {
+			err = p.cmd.Start()
+			if err != nil {
+				glog.Errorf("cmdRun() error starting %s: %v", p.name, err)
+				skipWait = true
 			}
-			done <- struct{}{}
-		}()
-	}
+		}
+		p.restartMutex.Unlock()
 
-	err = p.cmd.Start()
-	if err != nil {
-		glog.Errorf("cmdRun() error starting %s: %v", p.name, err)
-		return
-	}
+		<-done // goroutine is done
 
-	<-done
+		if restart && !skipWait {
+			err = p.cmd.Wait()
+			if err != nil {
+				glog.Errorf("cmdRun() error waiting for %s: %v", p.name, err)
+			}
+		}
 
-	err = p.cmd.Wait()
-	if err != nil {
-		glog.Errorf("cmdRun() error waiting for %s: %v", p.name, err)
-		return
+		time.Sleep(5 * time.Second) // Delay to prevent flooding restarts if startup fails
```

Comment:
Move this to a variable. also, 5 sec is too long between restarts. I would suggest 1 sec

Reply:
Okay, I can cut it to one second.  Okay, I can move the time to a constant.